File: usecase/src/main/java/br/org/restaurantepopular/usecase/category/DeleteCategoryUseCase.java
package br.org.restaurantepopular.usecase.category;

import br.org.restaurantepopular.usecase.core.IUseCase;

import java.util.Optional;

public class DeleteCategoryUseCase implements IUseCase<String, Optional> {

    private final ICategoryRepository categoryRepository;

    public DeleteCategoryUseCase(ICategoryRepository categoryRepository) {
        this.categoryRepository = categoryRepository;
    }

    @Override
    public Optional execute(String s) {
        categoryRepository.delete(s);
        return Optional.empty();
    }
}

File: usecase/src/main/java/br/org/restaurantepopular/usecase/category/DeleteCategoryUseCase.kt
package br.org.restaurantepopular.usecase.category

import br.org.restaurantepopular.usecase.core.IUseCase

class DeleteCategoryUseCase(
    private val categoryRepository: ICategoryRepository
) : IUseCase<String, Unit> {

    override fun execute(request: String) {
        categoryRepository.delete(request)
    }
}